examples/platform_channel/macos/Runner/PowerSource.swift

```diff
@@ -0,0 +1,78 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import Foundation
+import IOKit.ps
+
+enum PowerState {
+  case ac
+  case battery
+  case unknown
+}
+
+/// A convenience wrapper for an IOKit power source.
+class PowerSource {
```

Comment:
i'd recommend creating protocols for these types so it can be injected for unit tests